web/admin/apps.go

```diff
@@ -115,6 +115,8 @@ func (ar *Router) UpdateApp() http.HandlerFunc {
 			return
 		}
 
+		go ar.updateAllowedOrigins()
```

Comment:
1) Are you sure it will have time to finish its execution before **UpdateApp** terminates?
2) Do we really need 
```go
ar.originChecker.DeleteAll()
...
for _, a := range apps {
		ar.originChecker.AddRawURLs(a.RedirectURLs())
	}
```
if we've only updated one app?

Reply:
2. We cannot track whether an app was deleted or added, as I can see. We'll have to traverse the `origin` map or the `RedirectURLs()` anyway.

Reply:
2. I think, it'll be faster than searching for an old `RedirectURLs`.